File: file-templates/templates/MVVM/Cell.xctemplate/UICollectionViewCellObjective-C/___FILEBASENAME___.m
//___FILEHEADER___

#import "___FILEBASENAME___.h"
#import "___VARIABLE_productName___Model.h"

@interface ___FILEBASENAMEASIDENTIFIER___ ()
/// 容器视图
@property (nonatomic, strong) UIView *containerView;

@end
@implementation ___FILEBASENAMEASIDENTIFIER___

#pragma mark - private

- (void)onUpdateUIComponents {
    
}

- (void)onDataBindings {
    [self onUpdateUIComponents];
}

#pragma mark - setter/getter

- (void)setCellModel:(___VARIABLE_productName___Model *)cellModel {
    if (_cellModel != cellModel) {
        _cellModel = cellModel;
        [self onDataBindings];
    }
}

- (UIView *)containerView {
    if (!_containerView) {
        UIView *view = [[UIView alloc] initWithFrame:CGRectZero];
        [self.contentView addSubview:view];
        _containerView = view;
    }
    return _containerView;
}

@end

File: file-templates/templates/MVVM/Cell.xctemplate/UITableViewCellSwift/___FILEBASENAME___.swift
//___FILEHEADER___

import UIKit
import SnapKit

class ___FILEBASENAMEASIDENTIFIER___: ___VARIABLE_cocoaTouchSubclass___ {

    // MARK: - Porperties
    
    public var cellModel: ___VARIABLE_productName___Model? {
        didSet {
            onDataBindings()
        }
    }
    
    /// 容器视图
    private lazy var containerView: UIView = {
        let view = UIView()
        view.backgroundColor = .white
        return view
    }()
    
    // MARK: - UIComponents
    
    public func onInitUIComponents() {
        contentView.addSubview(containerView)
    }
    
    public func onLayoutUIComponents() {
        containerView.snp.makeConstraints { make in
            make.edges.equalTo(UIEdgeInsets(top: 0, left: 0, bottom: 0, right: 0))
        }
    }
 
    public func onUpdateUIComponents() {
        guard let cellModel = self.cellModel else { return }
        // Configure components for provider
    }
    
    public func onDataBindings() {
        onUpdateUIComponents()
    }
   

}
